### PR TITLE
Bookstore: Use Redux in React components

### PR DESCRIPTION
**Pull request by basitali111**
Project requirements
First of all - make sure that you keep state immutable in reducers!
Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
Import Redux Provider and your store in the main component of your React app.
Connect your app to the Redux store.
Make sure that list of books displays books from Redux store.
Use useSelector to consume the state.
In your React component responsible for adding new books, add functionality for "Add new" button click:
Import Redux useDispatch.
Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
Takes the inputs from the state and generate a unique id and add them to an object.
Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for removing books, add a functionality for "Remove" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).
In your React component responsible for categories, add a functionality for "Check status" button click:
Import Redux useDispatch.
Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -23,7 +24,9 @@ const bookReducer = (state = initialState, action) => {
     case ADD_BOOK:
       return [...state, action.payload];
     case DELETE_BOOK:
-      return [...state.filter((payload) => payload.id !== action.payload.id)];
+      console.log('before:', state);
+      console.log('after:', [...state.filter((book) => book.id !== action.payload.id)]);
```
Comment (codepantha) [inline src/redux/books/books.js]: - [ ] Please remove `console.logs` from your code as it makes your work look messy 🙏

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED ♻️ 

## Hi @basitali111 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ No linter errors
✔️ Using Redux in React components

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (basitali111) [reply src/redux/books/books.js]: > * [ ]  Please remove `console.logs` from your code as it makes your work look messy 🙏

Thanks for telling i forgot that ❤️‍🔥

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @basitali111 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**